pkg/networkmanager/wireguard.jsx

```diff
@@ -324,6 +348,22 @@ export function WireGuardDialog({ settings, connection, dev }) {
                                     id={idPrefix + '-allowedips-peer-' + i}
                                 />
                             </FormGroup>
+                            <FormGroup className='pf-m-6-col-on-md' label={_("Preshared key")} fieldId={idPrefix + '-presharedkey-peer-' + i} labelIcon={
```

Comment:
I think this needs some UI work: I suppose it doesn't make sense to specify *both* a public/private key pair *and* a PSK? So perhaps there should either be a "key mode" selector at the top (symmetric or asymmetric), or one field should be disabled if the other has a value.

Reply:
Both public/private key pair and PSK is used to make it more secure and quantum computer resistant. It is mentioned here https://www.wireguard.com/papers/wireguard.pdf.

Reply:
Found this in the docs (https://www.wireguard.com/protocol/#key-exchange-and-data-packets), I think this is more accessible than the pdf above.

> If an additional layer of symmetric-key crypto is required (for, say, post-quantum resistance), WireGuard also supports an optional pre-shared key that is mixed into the public key cryptography. When pre-shared key mode is *not* in use, the pre-shared key value used below is assumed to be an all-zero string of 32 bytes.